container_api/src/container.rs

```diff
@@ -125,13 +141,99 @@ impl Container {
         Ok(())
     }
 
+    pub fn spawn_network(&mut self) -> Result<String, HolochainError> {
+        let network_config = self
+            .config
+            .clone()
+            .network
+            .ok_or(HolochainError::ErrorGeneric(
+                "attempt to spawn network when not configured".to_string(),
+            ))?;
+
+        println!(
+            "spawn network (workdir: {})",
+            network_config.n3h_persistence_path
+        );
+        let SpawnResult {
+            kill,
+            ipc_binding,
+            p2p_bindings: _,
+        } = ipc_spawn(
+            "node".to_string(),
+            vec![format!(
+                "{}/packages/n3h/bin/n3h",
+                network_config.n3h_path.clone()
+            )],
+            network_config.n3h_persistence_path.clone(),
+            hashmap! {
+                String::from("N3H_MODE") => String::from("HACK"),
+                String::from("N3H_WORK_DIR") => network_config.n3h_persistence_path.clone(),
+                String::from("N3H_IPC_SOCKET") => String::from("tcp://127.0.0.1:*"),
+            },
+            true,
+        )
+        .map_err(|error| {
+            println!("!!!! {:?}", error);
+            HolochainError::ErrorGeneric(error.to_string())
+        })?;
+        self.network_child_process = kill;
+        println!("spawned with binding: {:?}", ipc_binding);
+        Ok(ipc_binding)
+    }
+
+    fn instance_network_config(
+        &self,
+        net_config: &NetworkConfig,
+    ) -> Result<JsonString, HolochainError> {
+        match self.network_ipc_uri {
+            Some(ref uri) => {
+                let tmp = JsonString::from(json!(
+                    {
+                        "backend_kind": "IPC",
+                        "backend_config": {
+                            "socketType": "zmq",
+                            "bootstrapNodes": net_config.bootstrap_nodes,
+                            "ipcUri": uri.clone()
+                        }
+                    }
+                ));
+
+                println!("NET CONFIG: {}", tmp.to_string());
+
+                Ok(tmp)
+            }
+            None => Err(HolochainError::ErrorGeneric(
+                "Network IPC URI missing".to_string(),
+            )),
+        }
+    }
+
     /// Tries to create all instances configured in the given Configuration object.
     /// Calls `Configuration::check_consistency()` first and clears `self.instances`.
     /// @TODO: clean up the container creation process to prevent loading config before proper setup,
     ///        especially regarding the signal handler.
     ///        (see https://github.com/holochain/holochain-rust/issues/739)
     pub fn load_config(&mut self) -> Result<(), String> {
         let _ = self.config.check_consistency()?;
+
+        // if there's no NetworkConfig then assume the mock network
```

Comment:
If this ought to be true we need to change the `None` case above in `fn instance_network_config`.